File: app/src/main/java/fr/igm/robotmissions/ui/robots/RobotAdapter.java
package fr.igm.robotmissions.ui.robots;

import android.content.Context;
import android.content.Intent;
import android.content.res.Resources;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import java.util.ArrayList;
import java.util.List;

import fr.igm.robotmissions.R;
import io.swagger.client.model.Robot;

public class RobotAdapter extends RecyclerView.Adapter<RobotAdapter.ViewHolder> {

    private List<Robot> robotList = new ArrayList<>();

    public RobotAdapter() {
        super();
    }

    public void setRobotList(List<Robot> robotList) {
        this.robotList = robotList;
        notifyDataSetChanged();
    }

    public class ViewHolder extends RecyclerView.ViewHolder{
        View cardView;
        TextView nameView;
        TextView batteryView;
        TextView isOnlineView;
        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            cardView = itemView.findViewById(R.id.robot_card_view);
            nameView = cardView.findViewById(R.id.robot_name_text);
            isOnlineView = cardView.findViewById(R.id.robot_is_online_text);
            batteryView = cardView.findViewById(R.id.robot_battery_text);
        }
    }
    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View itemView = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_robot, parent, false);
        return new RobotAdapter.ViewHolder(itemView);
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        Robot robot = robotList.get(position);

        // set detail intent click
        Context context = holder.cardView.getContext();


        // set item text
        holder.nameView.setText(robot.getName());
        Resources res = context.getResources();
        int onlineStrId = robot.isConnected() ? R.string.is_online : R.string.is_offline;
        holder.isOnlineView.setText(res.getString(onlineStrId));
        if (robot.getStatus() != null){
            holder.batteryView.setText(
                    String.format(
                            res.getString(R.string.robot_battery),
                            robot.getStatus().getBattery())
            );
        } else {
            holder.batteryView.setText(res.getString(R.string.robot_no_status));
        }
    }

    @Override
    public int getItemCount() {
        return robotList.size();
    }


}

File: app/src/main/java/fr/uge/robotsmissions/ui/robots/RobotAdapter.kt
package fr.uge.robotsmissions.ui.robots

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import fr.uge.robotsmissions.R
import fr.uge.robotsmissions.objects.Robot
import kotlinx.android.synthetic.main.item_robot.view.*

class RobotAdapter(private val robots: ArrayList<Robot>, val itemClickListener: View.OnClickListener) :
    RecyclerView.Adapter<RobotAdapter.ViewHolder>() {

    class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView), View.OnClickListener {

        val cardView = itemView.robot_card_view!!
        val nameView = cardView.robot_name_text!!
        val positionView = cardView.robot_ifc_position_name_text!!
        var onlineView = cardView.robot_is_online_text!!
        var batteryView = cardView.robot_battery_text!!

        val clickListener = itemView.setOnClickListener(this)

        override fun onClick(v: View?) {
            var position: Int = layoutPosition
            println(position)
        }

    }

    override fun getItemCount(): Int {
        return robots.size
    }

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RobotAdapter.ViewHolder {
        val viewItem = LayoutInflater.from(parent.context)
            .inflate(R.layout.item_robot, parent, false)
        return ViewHolder(viewItem)
    }

    override fun onBindViewHolder(holder: RobotAdapter.ViewHolder, position: Int) {
        val robot = robots[position]
        holder.cardView.setOnClickListener(itemClickListener)
        holder.cardView.tag = position
        holder.nameView.text = robot.name
        holder.positionView.text = "Position : ${robot.ifc_position_name}"
        holder.onlineView.text = when (robot.isOnline) {
            true -> "Status : Online"
            false -> "Status : Offline"
        }
        holder.batteryView.text = "Battery : ${robot.battery} %"

    }

    fun removeAt(position: Int) {
        robots.removeAt(position)
        notifyItemRemoved(position)
    }
}